### PR TITLE
test: raise coverage to 100% and cover remaining edge cases

### PR DESCRIPTION
**Pull request by brenopolanski**
## Summary

Bring runtime coverage to 100% and raise the Vitest thresholds to match.

**Before:**

Screenshot 2026-08-28 at 22 13 09

**After:**

Screenshot 2026-08-28 at 22 13 26

### CODE REVIEW

File: test/index.test.tsx [hunk 1]
```diff
@@ -416,6 +459,49 @@ it('falls back to console.error when onError is not provided', async () => {
...
+    <ImageEditor image="img-a" options={{ projectId: 1, theme: 'light' }} />
+  );
+  await flush();
+
+  rerender(
+    <ImageEditor image="img-b" options={{ projectId: 2, theme: 'dark' }} />
+  );
+  await flush();
+
+  expect(mockInstance.reset).not.toHaveBeenCalled();
```
Comment (lucasbesen) [inline test/index.test.tsx]: Could you update the remount test to use distinct editor instances and verify that `updateOptions` is not called on the old instance?

```ts
expect(oldInstance.updateOptions).not.toHaveBeenCalled();
```

The current test only checks reset, so the updateOptions guard is not covered.

Comment (brenopolanski) [reply test/index.test.tsx]: Good catch!

Updated the remount test to use distinct `oldInstance` and `newInstance` objects, and it now asserts `oldInstance.updateOptions` was never called.

Comment (lucasbesen) [reply test/index.test.tsx]: Thank you!
